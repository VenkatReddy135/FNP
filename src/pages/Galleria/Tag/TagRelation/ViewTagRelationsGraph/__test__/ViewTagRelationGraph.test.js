/* eslint-disable no-console */
import React from "react";
import { cleanup, render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import ViewTagRelationGraph from "../ViewTagRelationsGraph";

jest.mock("react-admin", () => ({
  useTranslate: () => (label) => label,
  useQueryWithStore: () => ({ loading: false }),
  useNotify: jest.fn(),
  useRedirect: () => () => console.log("redirected"),
}));

describe("Tag realtion Graph test cases", () => {
  const match = {
    params: {
      id: "anniversary",
    },
  };
  beforeEach(() => {
    render(<ViewTagRelationGraph match={match} />);
  });

  afterEach(() => {
    cleanup();
  });

  it("should render Tag Relation Graph UI component", () => {
    expect(screen).toBeTruthy();
  });

  it("should render element with given text", () => {
    expect(screen.getByText("view_tag_relation")).toBeInTheDocument();
  });

  it("should trigger close button ", () => {
    const consoleSpy = jest.spyOn(console, "log");
    const button = screen.getByText(/close/i);
    fireEvent.click(button);
    expect(consoleSpy).toHaveBeenCalledWith("redirected");
  });

  it("should match snapshot", () => {
    const tree = renderer.create(<ViewTagRelationGraph match={match} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
