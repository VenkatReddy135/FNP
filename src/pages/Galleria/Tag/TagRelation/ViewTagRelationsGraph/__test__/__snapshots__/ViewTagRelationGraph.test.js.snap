// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag realtion Graph test cases should match snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
>
  <h4
    className="MuiTypography-root MuiTypography-h4"
  >
    view_tag_relation
  </h4>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
    data-at-id="close"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      close
    </span>
  </button>
</div>
`;
